packages/blocks/src/api/factory.js

```diff
@@ -40,12 +41,17 @@ import { normalizeBlockType } from './utils';
  * @param {Object} attributes  Block attributes.
  * @param {?Array} innerBlocks Nested blocks.
  *
- * @return {Object} Block object.
+ * @return {(Object|boolean)} Block object or false if the block name is not a registered block.
  */
 export function createBlock( name, attributes = {}, innerBlocks = [] ) {
 	// Get the type definition associated with a registered block.
 	const blockType = getBlockType( name );
 
+	if ( undefined === blockType ) {
+		warning( `Block type '${ name }' is not registered.` );
+		return false;
```

Comment:
Making this non-blocking changes the expectation from someone calling this function. 

We do things like that in general in Gutenberg `compact( configs.map( config => createBlock( config.name, config.attributes, config.innerBlocks ) )`

With the current implementation, this will hide bugs and just move them down the road. IMO, we should throw an example here instead of calling `warning`.

Reply:
@youknowriad thanks for the review here! I have updated the PR to throw an exception.